CDCgov/prime-reportstream
# Correct Submission History to display "Received" and "Not Received" correctly

This PR corrects a display bug where the status for Submission History displays `Not Delivering` on reports that will be delivered but do not have Destinations calculated yet.

Test Steps:
1. Submit a report with data that you know will be delivered to at least 1 Receiver
2. Immediately after receiving the `SubmissionId`, use the `History API` to check the status of the report. It should display `Received`
3. Now submit a report with data that you know will not be delivered to any Receivers
4. Immediately after receiving the `SubmissionId`, use the `History API` to check the status of the report. It should display `Not Delivering`

## Changes
- Added a new var to track the total number of items that passed filtering on a submission
- Update `calculateStatus` to use the new var in calculation
- Added a few documentation fixes

## Linked Issues
- #9378
